monitor/src/main/java/edu/asu/diging/monitor/web/admin/ModifyAppController.java

```diff
@@ -0,0 +1,49 @@
+package edu.asu.diging.monitor.web.admin;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.monitor.core.model.IApp;
+import edu.asu.diging.monitor.core.service.IAppManager;
+import edu.asu.diging.monitor.web.admin.forms.AppForm;
+
+@Controller
+public class ModifyAppController {
+	
+	
+	@Autowired
+	private IAppManager appManager;
+
+	@RequestMapping(value="/admin/apps/{id}/modify", method= RequestMethod.GET)
+	public String show(Model model, @PathVariable("id") String id) {
+		model.addAttribute("appForm", appManager.getApp(id));
+		return "admin/apps/show";
+	}
+	
+	@RequestMapping(value="/admin/apps/{id}/modify", method=RequestMethod.POST)
+	public String update(@ModelAttribute AppForm appForm, @PathVariable("id") String id, RedirectAttributes redirectAttrs) {
+		IApp app = appManager.getApp(id);
+		app.setDescription(appForm.getDescription());
+		app.setExpectedReturnCodes(appForm.getExpectedReturnCodes());
+		app.setHealthUrl(appForm.getHealthUrl());
+		app.setName(appForm.getName());
+		app.setRetries(appForm.getRetries());
+		app.setTimeout(appForm.getTimeout());
+		app.setWarningReturnCodes(appForm.getWarningReturnCodes());
+		app.setPingInterval(appForm.getPingInterval());
+		app.setMethod(appForm.getMethod());
+		appManager.updateApp(app);
```

Comment:
since this is the same code as for adding an app (copying values from one object to the other), it should be in a helper class or in the manager class.